f/frizzle/comapeo/comapeo_observations.py

```diff
@@ -402,19 +419,16 @@ def handle_output(self, outputs):
 
             for row in rows:
                 try:
-                    # Wrap column names in single quotes
-                    columns = ", ".join(f'"{name}"' for name in row.keys())
-
-                    # Prepare the values list, converting None to 'NULL'
-                    values = [
-                        f"{value}" if value is not None else None
-                        for value in row.values()
-                    ]
+                    cols, vals = zip(*row.items())
```

Comment:
From this line to the rest of the file could use extra attention by reviewer.  In particular the handling of non-primitive (list and dict) values.

Reply:
Looks fine to me. And I tested the code with real CoMapeo data to be sure.